internal/utils/document_utils_test.go

```diff
@@ -43,6 +43,13 @@ func TestSplitDocSliceError(t *testing.T) {
 	assert.Empty(t, result)
 }
 
+func TestSplitDocSliceWithEmptyStruct(t *testing.T) {
```

Comment:
мы обычно объединяем тесты одной функции в одном тесте, но разделяем случаи через t.Run
```suggestion
func TestSplitDocSlice(t *testing.T) {    
    r.Run("SliceWithTail", ...)
    
    r.Run("WithEmptyStruct", ...)
```
и потом у тебя в функции 3 ретурна, а тестов всего 2, значит точно не 100% покрытие кода, т.е не все случаи учтены